### PR TITLE
play 2.5.x Anorm Doc update with using DI

### PR DESCRIPTION
**Pull request by abhisheknoldus**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [ ] Have you [squashed your commits](https://www.playframework.com/documentation/2.4.x/WorkingWithGit#Squashing-commits)?
* [ ] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [ ] Have you added tests for any changed functionality?

## Fixes

Fixes #6620
This PR is related Issue #6620 raised in Play framework. Here is the link : 
https://github.com/playframework/playframework/issues/6620

## Purpose

What does this PR do?
This PR update the scala Anorm documentation for play 2.5.x. In Play 2.5.x Database trait is there instead of DB object. So, This PR specify using Database trait members with the help of DI.

## Background Context

Why did you take this approach?

## References

Are there any relevant issues / PRs / mailing lists discussions?
https://github.com/playframework/playframework/issues/6620

### CODE REVIEW

File: docs/manual/working/scalaGuide/main/sql/code/ScalaAnorm.scala [hunk 1]
```diff
@@ -1,23 +1,14 @@
...
+package scalaGuide.main.sql
 
-import org.specs2.mutable.Specification
-import play.api.test._
-import play.api.test.Helpers._
+import play.api.db.Database
+import anorm._
+import com.google.inject.Inject
+//#playdb
+class ScalaAnorm  @Inject () (db: Database) {
```
Comment (wsargent) [inline docs/manual/working/scalaGuide/main/sql/code/ScalaAnorm.scala]: the imports should be available within the preview snippet

File: docs/manual/working/scalaGuide/main/sql/code/ScalaAnorm.scala [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
 
-object ScalaAnorm extends Specification {
-  "Code samples" title
+  db.withConnection { implicit connection =>
+    val result: Boolean = SQL("Select 1").execute()
 
-  "Anorm" should {
-    "be usable in play" in new WithApplication(FakeApplication(additionalConfiguration = inMemoryDatabase())) {
-      //#playdb
-      import anorm._
```
Comment (wsargent) [inline docs/manual/working/scalaGuide/main/sql/code/ScalaAnorm.scala]: Included imports

**Comment by abhisheknoldus** [comment]
@wsargent  requested changes has been done. Please check.

**Comment by majidhosseini1** [comment]
The document explaining how to use Anorm still does not mention the fact that play.api.db.DB has been deprecated (https://github.com/playframework/anorm/blob/master/docs/manual/working/scalaGuide/main/sql/ScalaAnorm.md)

**Pull request merged**